x-pack/solutions/search/plugins/enterprise_search/public/applications/enterprise_search_content/components/search_index/pipelines/pipelines.tsx

```diff
@@ -65,6 +65,28 @@ export const SearchIndexPipelines: React.FC = () => {
   const { makeRequest: revertPipeline } = useActions(RevertConnectorPipelineApilogic);
   const apiIndex = isApiIndex(index);
   const extractionDisabled = getContentExtractionDisabled(index);
+  const buttonRef = useRef<HTMLButtonElement | null>(null);
+  const pipelinesButton = Array.from(document.querySelectorAll('button')).find(
+    (btn) => btn.textContent?.trim() === 'Pipelines'
+  );
+
+  const onCloseDeleteModal = useCallback(() => {
+    closeDeleteModal();
+    setTimeout(() => {
+      if (buttonRef.current) {
+        buttonRef.current.focus();
+      }
+    }, 0);
+  }, [closeDeleteModal, buttonRef]);
+
+  const onDeletePipeline = useCallback(() => {
+    revertPipeline({ indexName });
+    setTimeout(() => {
+      if (pipelinesButton) {
+        pipelinesButton.focus();
+      }
+    }, 200);
+  }, [revertPipeline, pipelinesButton]);
```

Comment:
If we're going to query the DOM for the pipeline tab element we should do it here. We are setting the `id` for the tab in [search_index.tsx](https://github.com/elastic/kibana/blob/d05494781dbefafdf092221f30810fe9bfd97f01/x-pack/solutions/search/plugins/enterprise_search/public/applications/enterprise_search_content/components/search_index/search_index.tsx#L189), that id value should be passed down to the [Tab](https://github.com/elastic/eui/blob/main/packages/eui/src/components/tabs/tab.tsx#L109) and we should be able to query on that value like this:

```suggestion
  const onDeletePipeline = useCallback(() => {
    revertPipeline({ indexName });
    setTimeout(() => {
      const pipelinesButton = document. querySelector(`#${SearchIndexTabId.PIPELINES}`);
      if (pipelinesButton) {
        pipelinesButton.focus();
      }
    }, 200);
  }, [revertPipeline]);
```

You might also be able to create a ref and pass that to tab in `search_index.tsx`, but that may be overkill for what we need here.